lib/n_plus_one_control/rspec/matchers/perform_linear_number_of_queries.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# rubocop:disable  Metrics/BlockLength
+::RSpec::Matchers.define :perform_linear_number_of_queries do |slope: 1|
+  supports_block_expectations
+
+  match do |actual, *_args|
+    raise ArgumentError, "Block is required" unless actual.is_a? Proc
+
+    raise "Missing tag :n_plus_one" unless
+      @matcher_execution_context.respond_to?(:n_plus_one_populate)
+
+    populate = @matcher_execution_context.n_plus_one_populate
+
+    @matcher_execution_context.executor = NPlusOneControl::Executor.new(
+      population: populate,
+      matching: nil,
+      scale_factors: @factors
+    )
+
+    @queries = @matcher_execution_context.executor.call(&actual)
+
+    @queries.each_cons(2).all? do |pair|
```

Comment:
@palkan What do you think of this approach?

Reply:
Nice 👍 `#each_cons` is what I was missing